datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -850,6 +868,31 @@ pub async fn from_substrait_rel(
     }
 }
 
+/// Validates that the given Substrait schema matches the given DataFusion schema
+/// Returns true if the two schemas have the same qualified named fields with the same data types.
+/// Returns false otherwise.
+/// Ignores case when comparing field names
+///
+/// This code is equivalent to [DFSchema::equivalent_names_and_types] except that the field name
+/// checking is case-insensitive
+fn validate_substrait_schema(
+    datafusion_schema: &DFSchema,
+    substrait_schema: &DFSchema,
+) -> bool {
+    if datafusion_schema.fields().len() != substrait_schema.fields().len() {
+        return false;
+    }
+    let datafusion_fields = datafusion_schema.iter();
+    let substrait_fields = substrait_schema.iter();
+    datafusion_fields
+        .zip(substrait_fields)
+        .all(|((q1, f1), (q2, f2))| {
+            q1 == q2
+                && f1.name().to_lowercase() == f2.name().to_lowercase()
```

Comment:
I wasn't sure what the actual requirements around case-sensitivity are in DataFusion. I've opted for a more permissive check here, but this could potentially be configurable.